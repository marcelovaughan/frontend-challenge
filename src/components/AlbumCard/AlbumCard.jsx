import PropTypes from 'prop-types'
import React from 'react'
import moment from 'moment';

import styles from './AlbumCard.module.css'

const AlbumCard = ({ name, images, releaseDate }) => (
  <div className={styles.albumInfo}>
    <img className={styles.albumPhoto} width="100%" src={images && images.length > 0 && images[0].url} alt={name}/>
    <h3 title={name} className={styles.albumName}>{name}</h3>
    <h4 data-test="releaseDate" className={styles.albumDate}>{ moment(releaseDate).format("DD/MM/YYYY")}</h4>
  </div>
)

AlbumCard.propTypes = {
  name: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired,
  images: PropTypes.array,
}

export default AlbumCard
