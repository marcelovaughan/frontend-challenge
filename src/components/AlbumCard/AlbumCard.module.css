.albumInfo {
  border: 1px solid transparent;
  padding: 8px;
  font-size: 14px;
  text-align: left;
  margin: 0 0 10px;
  border-radius: 8px;
  background-color: rgba(255, 255, 255, 0.8);
  cursor: pointer;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  transition: background-color 0.5s, box-shadow 0.5s;
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.1);
  position: relative;
  display: grid;
}

.albumName {
  white-space: nowrap;
  width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 15px;
  margin-bottom: 20px;
  font-weight: bold;
  display: block;
  height: 20px;
}

.albumDate {
  font-size: 15px;
  font-weight: 100;
}

.albumPhoto {
  border-radius: 3px;
  margin-right: 10px;
  margin-bottom: 10px;
}
