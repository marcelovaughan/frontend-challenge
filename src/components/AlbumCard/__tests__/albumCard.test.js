import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16"

import AlbumCard from '../AlbumCard';

configure({ adapter: new Adapter() });

const mockProps = {
    name: 'SRV', 
    images: [{url: 'https://i.scdn.co/image/5bd95fbc961256c56560d838bd74d05be50701d4'}], 
    releaseDate: '2020-07-17'
}

describe('AlbumCard', () => {  
  test('should render AlbumCard', () => {
    const wrapper = shallow(<AlbumCard {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });  
});