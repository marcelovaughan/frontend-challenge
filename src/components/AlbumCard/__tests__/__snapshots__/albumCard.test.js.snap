// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumCard should render AlbumCard 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AlbumCard
    images={
      Array [
        Object {
          "url": "https://i.scdn.co/image/5bd95fbc961256c56560d838bd74d05be50701d4",
        },
      ]
    }
    name="SRV"
    releaseDate="2020-07-17"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="SRV"
          className="albumPhoto"
          src="https://i.scdn.co/image/5bd95fbc961256c56560d838bd74d05be50701d4"
          width="100%"
        />,
        <h3
          className="albumName"
          title="SRV"
        >
          SRV
        </h3>,
        <h4
          className="albumDate"
          data-test="releaseDate"
        >
          17/07/2020
        </h4>,
      ],
      "className": "albumInfo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "SRV",
          "className": "albumPhoto",
          "src": "https://i.scdn.co/image/5bd95fbc961256c56560d838bd74d05be50701d4",
          "width": "100%",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SRV",
          "className": "albumName",
          "title": "SRV",
        },
        "ref": null,
        "rendered": "SRV",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "17/07/2020",
          "className": "albumDate",
          "data-test": "releaseDate",
        },
        "ref": null,
        "rendered": "17/07/2020",
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="SRV"
            className="albumPhoto"
            src="https://i.scdn.co/image/5bd95fbc961256c56560d838bd74d05be50701d4"
            width="100%"
          />,
          <h3
            className="albumName"
            title="SRV"
          >
            SRV
          </h3>,
          <h4
            className="albumDate"
            data-test="releaseDate"
          >
            17/07/2020
          </h4>,
        ],
        "className": "albumInfo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "SRV",
            "className": "albumPhoto",
            "src": "https://i.scdn.co/image/5bd95fbc961256c56560d838bd74d05be50701d4",
            "width": "100%",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SRV",
            "className": "albumName",
            "title": "SRV",
          },
          "ref": null,
          "rendered": "SRV",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "17/07/2020",
            "className": "albumDate",
            "data-test": "releaseDate",
          },
          "ref": null,
          "rendered": "17/07/2020",
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
